addons/addon-base/packages/services/lib/s3-service.js

```diff
@@ -177,13 +177,37 @@ class S3Service extends Service {
       .promise();
   }
 
-  async createFolder(bucketName, folderName) {
+  async createPath(bucketName, folderName) {
     const params = {
       Bucket: bucketName,
       Key: folderName,
     };
     return this.api.putObject(params).promise();
   }
+
+  async clearPath(bucketName, dir) {
```

Comment:
Add unit test for this method?

Reply:
I don't think that would be necessary since the clearPath function is basically calling AWS APIs.

Reply:
In general, I don't think it really matters even if there is a single AWS API call- we should write a unit test regardless since the tests are added for regression as well(not for just the fact that things work right now). Having said that, there is a lot going on in that method ranging from list, delete and recursive clearPath call. Overall we need the unit tests to check if we are calling S3 with correct parameters, the base case works, and multiple iterations works (i.e. the recursion ends).

Reply:
Tests added in commit: 8eccd37803ee539e96dbc9839eab3bae57be8d5a